### PR TITLE
New rspace/plug in hot store

### PR DESCRIPTION
**Pull request by lukasz-golebiewski**
## Overview
Adapts RSpace for use with the newly created HotStore abstraction

### JIRA ticket:
Main part of https://rchain.atlassian.net/browse/RCHAIN-3142

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

File: rspace/src/main/scala/coop/rchain/rspace/nextgenrspace/SpaceMatcher.scala [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+/** The interface for RSpace
+  *
+  * @tparam C a type representing a channel
+  * @tparam P a type representing a pattern
+  * @tparam A a type representing an arbitrary piece of data
+  * @tparam R a type representing a match result
+  * @tparam K a type representing a continuation
+  */
+private[rspace] trait SpaceMatcher[F[_], C, P, A, R, K] extends ISpace[F, C, P, A, R, K] {
```
Comment (rabbitonweb) [inline rspace/src/main/scala/coop/rchain/rspace/nextgenrspace/SpaceMatcher.scala]: github sees this as a "new file" , better for maintainability if that was considered "moved and edit"

File: rspace/src/test/scala/coop/rchain/rspace/nextgenrspace/StorageTestsBase.scala [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+      }
+  }
+}
+
+import cats._
+import cats.effect._
+import coop.rchain.metrics.Metrics
+import coop.rchain.shared.Log
+
+trait TaskTests[C, P, A, R, K] extends StorageTestsBase[Task, C, P, R, K] {
```
Comment (rabbitonweb) [inline rspace/src/test/scala/coop/rchain/rspace/nextgenrspace/StorageTestsBase.scala]: where is it being used?

File: rspace/src/main/scala/coop/rchain/rspace/nextgenrspace/SpaceMatcher.scala (hunk 1 above)
Comment (lukasz-golebiewski) [reply rspace/src/main/scala/coop/rchain/rspace/nextgenrspace/SpaceMatcher.scala]: It wasn't moved, we need to preserve the original version for now

File: rspace/src/test/scala/coop/rchain/rspace/nextgenrspace/StorageTestsBase.scala (hunk 2 above)
Comment (lukasz-golebiewski) [reply rspace/src/test/scala/coop/rchain/rspace/nextgenrspace/StorageTestsBase.scala]:
```class InMemoryHotStoreStorageActionsTests
    extends InMemoryHotStoreTestsBase[Task]
    with TaskTests[String, Pattern, Nothing, String, StringsCaptor]
    with StorageActionsTests[Task]
    with StorageTestsBase[Task, String, Pattern, String, StringsCaptor]
```

**Pull request merged**